source/Dig/base.framework.entity.base.bmx

```diff
@@ -44,114 +44,102 @@ Import Brl.Map
 
 Type TEntityCollection
 	Field entriesID:TIntMap
-	Field entries:TMap = CreateMap()
+	Field entriesGUID:TStringMap {nosave}
 	Field entriesCount:int = -1
-	Field _entriesMapEnumerator:TNodeEnumerator {nosave}
 
 	Method Initialize:TEntityCollection()
 		'call Remove() for all objects so they can unregister stuff
 		'and tidy up in general
-		For local e:TEntityBase = EachIn entries.Values()
-			e.RemoveFromCollection(self)
-		Next
+		If entriesID
+			For local e:TEntityBase = EachIn entriesID.Values()
+				e.RemoveFromCollection(self)
+			Next
+		EndIf
+
+		If entriesID 
+			entriesID.Clear()
+			entriesID = Null
+		EndIf
+		If entriesGUID 
+			entriesGUID.Clear()
+			entriesGUID = Null
+		EndIf
 
-		entriesID.Clear()
-		entries.Clear()
 		entriesCount = -1
 
 		return self
 	End Method
 	
 	
-	Method CreateEntriesID()
-		If not entriesID 
-			entriesID = new TIntMap
-		Else
-			entriesID.Clear()
-		EndIf
+	Method GetEntriesID:TIntMap()
+		If not entriesID Then entriesID = new TIntMap
+		Return entriesID
+	End Method
+	
 
-		For local e:TEntityBase = EachIn entries.Values()
-			entriesID.Insert(e.GetID(), e)
-		Next
+	Method GetEntriesGUID:TStringMap()
+		If not entriesGUID 
+			entriesGUID = new TStringMap
+			For local e:TEntityBase = EachIn GetEntriesID().Values()
+				entriesGUID.Insert(e.GetGUID(), e)
+			Next
+		EndIf
+		
+		Return entriesGUID
 	End Method
 
 
 	Method GetByGUID:TEntityBase(GUID:String)
-		Return TEntityBase(entries.ValueForKey(GUID))
+		Return TEntityBase(GetEntriesGUID().ValueForKey(GUID))
 	End Method
 
 
 	Method Get:TEntityBase(ID:Int)
-		if not entriesID Then CreateEntriesID()
-
-		Return TEntityBase(entriesID.ValueForKey(ID))
+		Return TEntityBase(GetEntriesID().ValueForKey(ID))
 	End Method
 
 
 	Method GetCount:Int()
 		if entriesCount >= 0 then return entriesCount
 
 		entriesCount = 0
-		For Local base:TEntityBase = EachIn entries.Values()
+		For Local base:TEntityBase = EachIn GetEntriesID().Values()
 			entriesCount :+1
 		Next
 		return entriesCount
 	End Method
 
 
 	Method Add:int(obj:TEntityBase)
-		if not entriesID Then CreateEntriesID()
-		entriesID.Insert(obj.GetID(), obj)
-		entries.Insert(obj.GetGUID(), obj)
+		GetEntriesID().Insert(obj.GetID(), obj)
```

Comment:
Der Fehler passiert ja vorrangig beim hinzufügen. Wenn man also immer nur reinpackt, aber nie ein Remove macht, fällt es nie auf.

Mindestens in der Entwicklungsphase sollte eine automatische Konsistenzprüfung erfolgen (onDayBegins?)

Reply:
Ich habe das jetzt in ?debug builds beim "Add()" aktiviert. Die Konsistenzpruefung der Kollektionen an sich, ist ein eigenes Thema